libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"),
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_max_marginal_relevance_search(
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        """Test max marginal relevance search."""
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz", "bay", "bax", "baw", "bav"],
+            ids=["1", "2", "3", "4", "5", "6", "7"],
+            index_name=index_name,
+            set_name=set_name,
+        )
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=3
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo",
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
```

Comment:
**Security Issue**: The assertion checks if the length of mmr_output is 2 when fetch_k < k, which might not always hold true depending on the internal implementation of max_marginal_relevance_search. This could lead to flaky tests or false negatives in testing. <br> **Fix**: Add more robust error handling and conditional checks to ensure the test remains valid under various conditions. <br> **Code Suggestion**: 
 ```
 -        assert len(mmr_output) == 2
 +        assert len(mmr_output) <= min(3, fetch_k), "Check mmr_output length is less than or equal to the minimum of k and fetch_k"
 ```